src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -277,13 +277,13 @@ public HttpResponse doProceedEmpty() throws IOException, InterruptedException {
      * REST endpoint to abort the workflow.
      */
     @RequirePOST
-    public HttpResponse doAbort() throws IOException, InterruptedException {
+    public HttpResponse doAbort(Throwable cause) throws Exception{
```

Comment:
Right, the `do` part [is magic](https://github.com/jenkinsci/stapler/blob/master/docs/reference.adoc#action-method).

The `preAbortCheck` is likely irrelevant for a call to `stop`, and you can use `super.stop` rather than calling `onFailure` explicitly. So I think all that really needs to be changed is for https://github.com/julieheard/pipeline-input-step-plugin/blob/b07d21da1afb043a5eeedec8d575a13c04773702/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java#L141 to call `postSettlement`, and for `stop` to otherwise just set `outcome` and call the super method.